sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/jdbc/JdbcConnectionStringTest.java

```diff
@@ -0,0 +1,121 @@
+package com.azure.spring.cloud.autoconfigure.jdbc;
+
+import com.azure.spring.cloud.autoconfigure.implementation.jdbc.AzureUnsupportedDatabaseTypeException;
+import com.azure.spring.cloud.autoconfigure.implementation.jdbc.DatabaseType;
+import com.azure.spring.cloud.autoconfigure.implementation.jdbc.JdbcConnectionString;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.azure.spring.cloud.autoconfigure.implementation.jdbc.JdbcConnectionString.INVALID_CONNECTION_STRING_FORMAT;
+import static com.azure.spring.cloud.autoconfigure.implementation.jdbc.JdbcConnectionString.INVALID_PROPERTY_PAIR_FORMAT;
+import static com.azure.spring.cloud.autoconfigure.implementation.jdbc.JdbcPropertyConstants.MYSQL_AUTH_PLUGIN_CLASS_NAME;
+import static com.azure.spring.cloud.autoconfigure.implementation.jdbc.JdbcPropertyConstants.NONE_VALUE;
+import static com.azure.spring.cloud.autoconfigure.implementation.jdbc.JdbcPropertyConstants.PROPERTY_NAME_MYSQL_AUTHENTICATION_PLUGINS;
+import static com.azure.spring.cloud.autoconfigure.implementation.jdbc.JdbcPropertyConstants.PROPERTY_NAME_MYSQL_DEFAULT_AUTHENTICATION_PLUGIN;
+import static com.azure.spring.cloud.autoconfigure.implementation.jdbc.JdbcPropertyConstants.PROPERTY_NAME_MYSQL_SSL_MODE;
+import static com.azure.spring.cloud.autoconfigure.implementation.jdbc.JdbcPropertyConstants.PROPERTY_NAME_MYSQL_USE_SSL;
+import static com.azure.spring.cloud.autoconfigure.implementation.jdbc.JdbcPropertyConstants.PROPERTY_VALUE_MYSQL_SSL_MODE;
+import static com.azure.spring.cloud.autoconfigure.implementation.jdbc.JdbcPropertyConstants.PROPERTY_VALUE_MYSQL_USE_SSL;
+import static com.azure.spring.cloud.autoconfigure.jdbc.JdbcPropertiesBeanPostProcessor.DEFAULT_ENHANCED_PROPERTIES;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrowsExactly;
+
+class JdbcConnectionStringTest {
+    private static final String MYSQL_AUTH_PLUGIN_PROPERTY = PROPERTY_NAME_MYSQL_AUTHENTICATION_PLUGINS + "=" + MYSQL_AUTH_PLUGIN_CLASS_NAME;
+    private static final String MYSQL_DEFAULT_PLUGIN_PROPERTY = PROPERTY_NAME_MYSQL_DEFAULT_AUTHENTICATION_PLUGIN + "=" + MYSQL_AUTH_PLUGIN_CLASS_NAME;
+    private static final String MYSQL_SSL_MODE_PROPERTY = PROPERTY_NAME_MYSQL_SSL_MODE + "=" + PROPERTY_VALUE_MYSQL_SSL_MODE;
+    private static final String MYSQL_USE_SSL_PROPERTY = PROPERTY_NAME_MYSQL_USE_SSL + "=" + PROPERTY_VALUE_MYSQL_USE_SSL;
+
+    @Test
```

Comment:
Do we have a test case here to cover the `AzureUnsupportedDatabaseTypeException`?